audio/wav.go

```diff
@@ -113,9 +115,28 @@ func NewWAVFromData(data []byte) (*WAV, error) {
 
 	// hOff is the header offset. Even though the header length is actually
 	// 44, we find where the data begins by looking for the letters
-	// "data" which is from bytes 36 to 39. The variable i at this point
-	// pointing to right past the "data" letters
-	hOff := i - 40
+	// "RIFF" which is from bytes 0 to 3. The variable i at this point
+	// pointing to right past the "RIFF" letters
+	hOff := i - 4
+
+	if ((len(data) - hOff - 44) < 0) {
+		return nil, errors.NewFromErrorCode(errors.WAVCorruptFile)
+	}
+
+	// Verifies that "WAVE" letters exist in big endian form
+	if (data[hOff+8] != 'W' || data[hOff+9] != 'A' || data[hOff+10] != 'V' || data[hOff+11] != 'E') {
+		return nil, errors.NewFromErrorCode(errors.WAVCorruptFile)
```

Comment:
These should all specify the `Info` field so that the developer can know exactly what you found that was corrupt.